src/domain/networking/packets/EntityData.ts

```diff
@@ -9,29 +9,1932 @@
+                            ? this.#_UNKNOWN_OCTCODE_LENGTH
+                            : maxBytes - 1;
+
+                        if (newMaxBytes === this.#_OVERFLOWED_OCTCODE_BUFFER) {
+                            result += this.#_OVERFLOWED_OCTCODE_BUFFER;
+                            break;
+                        }
+                    }
+                    return result;
+                };
+
+                const numberOfThreeBitSectionsInStream
+                = numberOfThreeBitSectionsInCode(entityData, entityDataPosition, entityData.byteLength);
+
+                const bytesRequiredForCodeLength = (threeBitCodes: number): number => {
+                // C++ size_t bytesRequiredForCodeLength(unsigned char threeBitCodes)
+
+                    if (threeBitCodes === 0) {
+                        return 1;
+                    }
+                    return 1 + Math.ceil(threeBitCodes * 3 / 8.0);
+                };
+
+                const octalCodeBytes = bytesRequiredForCodeLength(numberOfThreeBitSectionsInStream);
+                entityDataPosition += octalCodeBytes;
+
+                // 1 represents sizeof(unsigned char) in the C++ code.
+                if (entityData.byteLength - entityDataPosition < 1) {
+                    break;
+                }
+
+                // WEBRTC TODO: Read and use the variable colorInPacketMask as in C++ method Octree::readElementData.
+                entityDataPosition += 1;
+
+                // WEBRTC TODO: Do not hardcode value.
+                // eslint-disable-next-line
+                // @ts-ignore
+                const bytesForMask = 2;
+                entityDataPosition += bytesForMask;
+
+                // WEBRTC TODO: Address further C++ code.
+
+                // 2 represents sizeof(numberOfEntities) in the C++ code.
+                if (entityData.byteLength - entityDataPosition < 2) {
+                    break;
+                }
+
+                const numberOfEntities = entityData.getUint16(entityDataPosition, UDT.LITTLE_ENDIAN);
+                entityDataPosition += 2;
+
+                if (entityData.byteLength < numberOfEntities * this.#_MINIMUM_HEADER_BYTES) {
+                    break;
+                }
+
+                for (let i = 0; i < numberOfEntities; i++) {
+                    const entityItemID = new Uuid(entityData.getBigUint128(entityDataPosition, UDT.BIG_ENDIAN));
+                    entityDataPosition += 16;
+
+                    const entityTypeCodec = new ByteCountCoded();
+                    let encodedData = new DataView(entityData.buffer, entityData.byteOffset + entityDataPosition);
+                    entityDataPosition += entityTypeCodec.decode(encodedData, encodedData.byteLength);
+
+                    const entityType = entityTypeCodec.data;
+
+                    if (entityType !== EntityTypes.Model) {
+                        console.error("Entity is not a Model entity");
+                        return [];
```

Comment:
This is going to return from the read() method immediately without any entity data, even if Model entities have been read. Instead, it should skip over the current entity's data and continue on with reading the next.

Reply:
The issue then is, as far as I can tell, there's no way to know how many bytes to skip.

There's an `EntityItem::expectedBytes()` (see link below) but it doesn't help. It's actually more of a `minimumExpectedBytes()` than a true count of the expected bytes.
https://github.com/vircadia/vircadia/blob/master/libraries/entities/src/EntityTree.cpp#L197

Nor does it appear to be encoded anywhere in the data.

I could break from the loop maybe. And return what has already been extracted. It would cut at the first non Model entity.

Reply:
Possible code:
```
                    if (entityType !== EntityTypes.Model) {
                        dataPosition += 2;  // sectionLength bytes
                        dataPosition += sectionLength;
                        continue;
                    }

```